lib/jwt/decode.rb

```diff
@@ -2,6 +2,7 @@
 
 require 'json'
 require 'jwt/x5c_key_finder'
+require 'jwt/x5t_key_finder'
```

Comment:
First very high level comment is that I think i would like to move all the keyfinders to their own namespace. `JWT::KeyFinders::X5t` in this case...

Reply:
That makes sense. What do you think about moving all the key finder logic to `JWT::JWK::KeyFinder`? We can abstract the logic to find key in this class. The only problem may be keeping `x5c` logic backward-compatible 🤔

Reply:
I think I have an idea. I'll work on it tomorrow and commit. Then you can review and see whether it could work or not.

Reply:
Also if the class/module would be compatible with the new Token classes it would be cool